frontend/src/components/tasks.jsx

```diff
@@ -0,0 +1,180 @@
+  }, [date]);
+
+  const months = ['January', 'February', 'March', 'April', 'May', 'June',
+    'July', 'August', 'September', 'October', 'November', 'December',
+  ];
+
+  const setDateBack = () => {
+    const yesterday = new Date(date);
+    yesterday.setDate(yesterday.getDate() - 1);
+    setDate(yesterday);
+  };
+
+  const setDateForward = () => {
+    const tomorrow = new Date(date);
+    tomorrow.setDate(tomorrow.getDate() + 1);
+    setDate(tomorrow);
+  };
+
+  const getChecked = (title) => {
+    // returns if a task is checked, false if not in checked map
+    const isChecked = checked.get(title);
+    return isChecked || false;
+  };
+
+  const getNumComplete = () => {
+    let numComplete = 0;
+    checked.forEach((key, value) => {
+      if (value) {
+        numComplete += 1;
+      }
+    });
+    return numComplete;
+  };
+
+  const onCheckedChange = (title, isChecked) => {
+    const temp = new Map(checked);
+    if (isChecked) {
+      temp.set(title, isChecked);
+    } else {
+      temp.delete(title);
+    }
+    setChecked(temp);
+  };
+
+  useEffect(() => {
+    // determine completion progress when checked is changed
+    const numTasks = tasks.length;
+    const numComplete = getNumComplete();
+    setCompletion(Math.floor((numComplete / numTasks) * 100));
+  }, [checked]);
+
+  return (
+    <div>
+      <Taskbar />
+      <Grid
+        className="main"
+        container
+        direction="row"
+      >
+        <Grid item xs={3}>
+          <LocalizationProvider dateAdapter={AdapterDateFns}>
+            <CalendarPicker date={date} onChange={(newDate) => setDate(newDate)} color="secondary" />
+          </LocalizationProvider>
+        </Grid>
+        <Grid
+          item
+          xs={3}
+          container
+          direction="column"
+          justifyContent="flex-start"
+          alignItems="center"
+        >
+          <Grid item sx={{ paddingBottom: 10 }}>
+            <Grid
+              className="dateSelector"
+              container
+              direction="row"
+              justifyContent="space-evenly"
+              alignItems="center"
+            >
+              <Grid item>
+                <IconButton
+                  aria-label="back"
+                  sx={{ color: '#1d4d71' }}
+                  size="large"
+                  onClick={setDateBack}
+                >
+                  <ArrowBackIosIcon />
+                </IconButton>
+              </Grid>
+              <Grid item>
+                <Typography
+                  variant="h4"
+                  style={{ fontWeight: 600 }}
+                  sx={{ color: '#1d4d71' }}
+                >
+                  {`${months[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`}
+                </Typography>
+              </Grid>
+              <Grid item>
+                <IconButton
+                  aria-label="forward"
+                  sx={{ color: '#1d4d71' }}
+                  size="large"
```

Comment:
When resizing the page components overlap. Let's collapse the calendar first and lets also make the < Date > component text a little smaller. and auto adjust size for circular progress along with the date and tasks

Reply:
see latest commit